File: CH01_CodeSmells/Solutions/MessageChains.cs
namespace RefactoringToDesignPatterns.CH01_CodeSmells.Solutions
{
    internal class MessageChains
    {
        public class Engine
        {
            public Cylinder GetCylinder()
            {
                return new Cylinder();
            }
        }

        public class Cylinder
        {
            public string GetSize()
            {
                return "2.0L";
            }
        }

        public class Car
        {
            public string GetCylinderSize()
            {
                Engine engine = new Engine();
                return engine.GetCylinder().GetSize();
            }
        }

        // Przykład wywołania
        //Car car = new Car();
        //string cylinderSize = car.GetCylinderSize();
    }
}

File: solutions_python/CH01_CodeSmells/MessageChains.py
class Cylinder:
    @staticmethod
    def get_size() -> str:
        return "2.0L"

class Engine:
    @staticmethod
    def get_cylinder() -> Cylinder:
        return Cylinder
    
class Car:
    @staticmethod        
    def get_cylinder_size(engine: Engine):
        return (
            engine
            .get_cylinder()
            .get_size()
        )
    

if __name__ == "__main__":
    car = Car()
    cylinder_size = (
        car.get_cylinder_size(engine=Engine())
    )
    print(cylinder_size)